flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/rules/logical/PushFilterIntoTableSourceScanRule.scala

```diff
@@ -83,6 +84,14 @@ class PushFilterIntoTableSourceScanRule extends RelOptRule(
 
     val newTableSource = filterableSource.applyPredicate(remainingPredicates)
 
+    if (remainingPredicates.size() > 0
+      && newTableSource.asInstanceOf[FilterableTableSource[_]].isFilterPushedDown
+      && newTableSource.explainSource().equals(scan.tableSource.explainSource())) {
+      throw new TableException("Failed to push filter into table source! "
```

Comment:
throwing `TableException` here. do you think we should make this change a soft one to only generate a warning and still let the planner to go through?

Reply:
I think throw exception is fine, since it's not easy to see the logs which generated during optimization.